File: backend/app/admin/service/internal/service/department.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"github.com/tx7do/go-utils/trans"
	pagination "github.com/tx7do/kratos-bootstrap/api/gen/go/pagination/v1"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-admin/app/admin/service/internal/data"

	adminV1 "kratos-admin/api/gen/go/admin/service/v1"
	userV1 "kratos-admin/api/gen/go/user/service/v1"

	"kratos-admin/pkg/middleware/auth"
)

type DepartmentService struct {
	adminV1.DepartmentServiceHTTPServer

	log *log.Helper

	uc *data.DepartmentRepo
}

func NewDepartmentService(uc *data.DepartmentRepo, logger log.Logger) *DepartmentService {
	l := log.NewHelper(log.With(logger, "module", "department/service/admin-service"))
	return &DepartmentService{
		log: l,
		uc:  uc,
	}
}

func (s *DepartmentService) List(ctx context.Context, req *pagination.PagingRequest) (*userV1.ListDepartmentResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *DepartmentService) Get(ctx context.Context, req *userV1.GetDepartmentRequest) (*userV1.Department, error) {
	return s.uc.Get(ctx, req)
}

func (s *DepartmentService) Create(ctx context.Context, req *userV1.CreateDepartmentRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.CreateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Create(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *DepartmentService) Update(ctx context.Context, req *userV1.UpdateDepartmentRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.UpdateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Update(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *DepartmentService) Delete(ctx context.Context, req *userV1.DeleteDepartmentRequest) (*emptypb.Empty, error) {
	if _, err := s.uc.Delete(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

File: frontend/apps/admin/src/rpc/service/department.ts
import type { DepartmentService } from '#/rpc/api/admin/service/v1/i_department.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';
import type {
  CreateDepartmentRequest,
  DeleteDepartmentRequest,
  Department,
  GetDepartmentRequest,
  ListDepartmentResponse,
  UpdateDepartmentRequest,
} from '#/rpc/api/user/service/v1/department.pb';

import { requestClient } from '#/rpc/request';

/** 部门管理服务 */
class DepartmentServiceImpl implements DepartmentService {
  async Create(request: CreateDepartmentRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/departments', request);
  }

  async Delete(request: DeleteDepartmentRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/departments/${request.id}`);
  }

  async Get(request: GetDepartmentRequest): Promise<Department> {
    return await requestClient.get<Department>(`/departments/${request.id}`);
  }

  async List(request: PagingRequest): Promise<ListDepartmentResponse> {
    return await requestClient.get<ListDepartmentResponse>('/departments', {
      params: request,
    });
  }

  async Update(request: UpdateDepartmentRequest): Promise<Empty> {
    const id = request.data?.id;
    if (request.data !== null && request.data !== undefined) {
      request.data.id = undefined;
    }
    return await requestClient.put<Empty>(`/departments/${id}`, request);
  }
}

export const defDepartmentService = new DepartmentServiceImpl();
